Automattic/jetpack
# Move the Protect plugin models into it's own package.

In this PR we're moving the models (classes) used in the Protect plugin into their own package so that they can be shared.
Ultimately (next) we will be creating a `protect-status` package that will need these models, as well as the Protect plugin itself needs the models (therefore the shared Protect_Models package).







## Proposed changes:

* Introduce new `protect-models` package. (Used the `jetpack generate` command.)
* Created mirror repo at: https://github.com/Automattic/jetpack-protect-models

### Other information:


## Jetpack product discussion



This PR is part of the _PT: Add value to the active Protect product card on My Jetpack_ project: pbNhbs-aP6-p2


## Does this pull request change what data or activity we track or use?



No

## Testing instructions:






* Checkout this branch locally
* Run `jetpack install packages/protect-models` and `jetpack build packages/protect-models`
* From the Protect plugin folder `cd projects/plugins/protect` run `composer require automattic/jetpack-protect-models`
* Open `projects/plugins/protect/src/class-status.php` and under the namespace declaration add the following `use` statements: (to lines 10 & 11)
 ```
use Automattic\Jetpack\Protect_Models\Extension_Model;
use Automattic\Jetpack\Protect_Models\Status_Model;
```
* Open `projects/plugins/protect/src/class-protect-status.php` and under the namespace declaration add the following `use` statements: (to lines 13 & 14 & 15)
 ```
use Automattic\Jetpack\Protect_Models\Extension_Model;
use Automattic\Jetpack\Protect_Models\Status_Model;
use Automattic\Jetpack\Protect_Models\Threat_Model;
```
* Open `projects/plugins/protect/src/class-scan-status.php` and under the namespace declaration add the following `use` statements: (to lines 13 & 14 & 15)
 ```
use Automattic\Jetpack\Protect_Models\Extension_Model;
use Automattic\Jetpack\Protect_Models\Status_Model;
use Automattic\Jetpack\Protect_Models\Threat_Model;
```
* Run `jetpack install plugins/protect && jetpack build plugins/protect`
* Go to `/wp-admin/admin.php?page=jetpack-protect` and confirm it loads Protect plugin page as expected and it displays scan results.
* To run unit tests, go to `projects/packages/protect_models` and run `composer test-php`
